python/doc/source/timestamps.rst

```diff
@@ -0,0 +1,433 @@
+All About Timestamps (work in progress)
```

Comment:
This is a big document. I'd like to see if we can make this about 50% as long or less. I will review in more detail as soon as I can and make some comments to help

Reply:
It is a big document. It's pretty long right now because there are quite bit of concepts to clarify, about 50% of the doc is about concepts and the other half is about Arrow <-> Spark.